FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,117 @@
         {
-            CoordinateFirst = coordinateFirst;
-            CoordinateSecond = coordinateSecond;
-            CoordinateThird = coordinateThird;
+            FirstCoordinate = coordinateFirst;
+            SecondCoordinate = coordinateSecond;
+            ThirdCoordinate = coordinateThird;
         }
+        
+        public Vector(Vector vector) : this(vector.FirstCoordinate, vector.SecondCoordinate, vector.ThirdCoordinate) { }
 
-        public Vector(Vector vector) : this()
+        public double CalculateAngleBetweenVectors(Vector vector)
         {
-            CoordinateFirst = vector.CoordinateFirst;
-            CoordinateSecond = vector.CoordinateSecond;
-            CoordinateThird = vector.CoordinateThird;
+            var angle = VectorHelper.CalculateAngleBetweenVectors(this, vector);
+            return Math.Round(angle, 3);
         }
 
-        public override string ToString()
-        {
-            return $"({CoordinateFirst}, {CoordinateSecond}, {CoordinateThird})";
-        }
+        public double CalculateModuleVector(Vector vector) => VectorHelper.CalculateModuleVector(vector);
+        
+        public override string ToString() => $"({FirstCoordinate}, {SecondCoordinate}, {ThirdCoordinate})";
 
-        public static Vector operator +(Vector vector1, Vector vector2)
+        public override int GetHashCode()
         {
-            var result = new Vector(vector1);
+            var hashCodeProperties = new int[]
+                {
+                    FirstCoordinate.GetHashCode(),
+                    SecondCoordinate.GetHashCode(),
+                    ThirdCoordinate.GetHashCode()
+                };
 
-            result.CoordinateFirst += vector2.CoordinateFirst;
-            result.CoordinateSecond += vector2.CoordinateSecond;
-            result.CoordinateThird += vector2.CoordinateThird;
+            var result = hashCodeProperties[0];
 
-            return result;
-        }
-
-        public static Vector operator -(Vector vector1, Vector vector2)
-        {
-            var result = new Vector(vector1);
-
-            result.CoordinateFirst -= vector2.CoordinateFirst;
-            result.CoordinateSecond -= vector2.CoordinateSecond;
-            result.CoordinateThird -= vector2.CoordinateThird;
+            for (int i = 1; i < hashCodeProperties.Length; i++)
+            {
+                result = unchecked(result * 397) ^ hashCodeProperties[i];
+            }
 
             return result;
         }
 
-        public static Vector operator *(double numeric, Vector vector)
+        public override bool Equals(object obj) => this.Equals(obj as Vector);
+  
+        public bool Equals(Vector other)
         {
-            return new Vector(vector.CoordinateFirst * numeric, vector.CoordinateSecond * numeric, vector.CoordinateThird * numeric);
+            if (ReferenceEquals(this, other)) return true;
+            if ((object)other == null) return false;
```

Comment:
It would be better to check whether **other** is _null_ or not before any other validations, don't you think so?

Reply:
By the way, why do you cast **other** to _object_?